### PR TITLE
[HZ-317] Add ClientConsole entrypoint into HazelcastCommandLine

### PR DESCRIPTION
**Pull request by ufukyilmaz**
This PR adds a new `hazelcast console` command to
`HazelcastCommandLine` to start the `ClientConsoleApp`.

Also, I replaced the line reader of client console app with jline reader
to add clear terminal, history like functionalities to this application
and made some cleanup on commands which was not working.

Additionally, I extracted `SqlConsole` to a separate class from `HazelcastCommandLine`.

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [ ] Add `Add to Release Notes` label if changes should be mentioned in release notes or `Not Release Notes content` if changes are not relevant for release notes
- [ ] Request reviewers if possible
- [ ] New public APIs have `@Nonnull/@Nullable` annotations
- [ ] New public APIs have `@since` tags in Javadoc
- [ ] Send backports/forwardports if fix needs to be applied to past/future releases

### CODE REVIEW

**Comment by ufukyilmaz** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java [hunk 1]
```diff
@@ -267,19 +295,38 @@ public void run() {
...
             println("echo: " + echo);
+        } else if (equalsIgnoreCase(first, "clear")) {
+            lineReader.getTerminal().puts(InfoCmp.Capability.clear_screen);
+        } else if (equalsIgnoreCase(first, "history")) {
+            History hist = lineReader.getHistory();
+            ListIterator<History.Entry> iterator = hist.iterator();
+            while (iterator.hasNext()) {
+                History.Entry entry = iterator.next();
+                if (iterator.hasNext()) {
+                    String entryLine = new StringBuilder()
```
Comment (mdumandag) [inline hazelcast/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java:310]: I see that we use something like 
```java
new AttributedStringBuilder().style(AttributedStyle.BOLD.foreground(PRIMARY_COLOR))....
```
for the history command in `HazelcastCommandLine`. Would it be better if we use the same approach here?

File: hazelcast/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java [hunk 2]
```diff
@@ -89,6 +104,8 @@
     private static final int BYTE_TO_BIT = 8;
     private static final int LENGTH_BORDER = 4;
 
+    private static final int COLOR = 12;
```
Comment (mdumandag) [inline hazelcast/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java:107]: Just curious, what color is that?

**Review by mdumandag — COMMENTED** [review]
Looks good overall, left some minor comments

File: hazelcast/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java [hunk 3]
```diff
@@ -267,19 +295,38 @@ public void run() {
         } else if ("silent".equals(first)) {
             silent = Boolean.parseBoolean(args[1]);
         } else if ("shutdown".equals(first)) {
-            hazelcast.getLifecycleService().shutdown();
+            client.getLifecycleService().shutdown();
         } else if ("echo".equals(first)) {
             echo = Boolean.parseBoolean(args[1]);
             println("echo: " + echo);
+        } else if (equalsIgnoreCase(first, "clear")) {
```
[the remark this answers is not in the document]
Comment (ufukyilmaz) [reply hazelcast/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java:283]: Made all of these general commands case insensitive

File: hazelcast/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java [hunk 4]
```diff
@@ -395,9 +423,9 @@ public void run() {
             handleAtomicNumberGet(args);
         } else if ("a.set".equals(first)) {
             handleAtomicNumberSet(args);
-        } else if ("a.inc".equals(first)) {
+        } else if ("a.incrementAndGet".equals(first)) {
             handleAtomicNumberInc(args);
-        } else if ("a.dec".equals(first)) {
+        } else if ("a.decrementAndGet".equals(first)) {
```
Comment (ramizdundar) [inline hazelcast/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java:428]: In my opinion keeping short forms would be better (we can still have long ones)

File: hazelcast/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java [hunk 5]
```diff
@@ -264,22 +275,39 @@ public void run() {
...
+                } else {
+                    // remove the "history" command from the history
+                    iterator.remove();
+                    hist.resetIndex();
+                }
+            }
+        } else if (equalsIgnoreCase(first, "ns")) {
             handleNamespace(args);
-        } else if ("whoami".equals(first)) {
-            handleWhoami();
```
Comment (ramizdundar) [inline hazelcast/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java:277]: Why is this removed?

File: hazelcast/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java [hunk 6]
```diff
@@ -1386,160 +1414,214 @@ protected void handleHelp(String command) {
...
-
+        printlnBold("Executor Service commands:");
+        println("  execute <echo-input>                    executes an echo task on random member");
+        println("  executeOnKey <echo-input> <key>         executes an echo task on the member that");
+        println("                                            owns the given key");
+        println("  executeOnMembers <echo-input>           executes an echo task on all of the members");
+        println("  executeOnMembers <echo-input>           executes an echo task on the member with");
+        println("                     <memberIndex>          given index");
+        println("  e<threadcount>.simulateLoad             simulates load on executor with given number");
+        println("          <task-count> <delaySeconds>       of thread (e1..e16)");
```
Comment (ramizdundar) [inline hazelcast/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java:1537]:
```suggestion
        println("  executeOnMembers <echo-input> <memberIndex>");
        println("    executes an echo task on the member with given index");
        println("");
        println("  e<threadcount>.simulateLoad  <task-count> <delaySeconds>");
        println("    simulates load on executor with given number of thread (e1..e16)");
```

maybe this format would be better for the commands that doesn't fit

**Review by ramizdundar — APPROVED** [review]
Minor comments

File: hazelcast/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java (hunk 4 above)
Comment (ufukyilmaz) [reply hazelcast/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java:428]: There are also variants of these, such as `getAndDecrement` and `getAndIncrement`, I chose this format so that they would not confused with them.

File: hazelcast/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java (hunk 5 above)
Comment (ufukyilmaz) [reply hazelcast/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java:277]: This command was not working

File: hazelcast/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java (hunk 6 above)
Comment (ufukyilmaz) [reply hazelcast/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java:1537]: I think it is better to align descriptions, so I'll keep it as it is
